p2p/src/handshake.rs

```diff
@@ -253,26 +254,10 @@ pub mod peer_state {
     );
 
     /// Peer in disconnected state.
-    #[derive(Debug)]
-    pub struct Disconnected(peer::Id);
+    #[derive(Debug, IdOrdEqHash)]
+    pub struct Disconnected(#[id] peer::Id);
 
     /// Peer in broken state.
-    #[derive(Debug)]
-    pub struct Broken(peer::Id, crate::Error);
-
-    macro_rules! impl_id {
-        [ $($ty:ty),* ] => {
-            $(
-            impl iroha_data_model::Identifiable for $ty {
-                type Id = iroha_data_model::peer::Id;
-
-                fn id(&self) -> &Self::Id {
-                    &self.0
-                }
-            }
-            )*
-        }
-    }
-
-    impl_id![Connecting, ConnectedTo, ConnectedFrom, Disconnected, Broken];
+    #[derive(Debug, IdOrdEqHash)]
+    pub struct Broken(#[id] peer::Id, crate::Error);
```

Comment:
@appetrosyan are we ok with this change?

Reply:
LGTM.